src/pages/index.js

```diff
@@ -4,91 +4,195 @@ import "../pages/index.css";
 import UserInfo from "../components/UserInfo.js";
 import Section from "../components/Section.js";
 import ImagePopup from "../components/PopupWithImage.js";
+import PopupConfirm from "../components/PopupConfirm.js";
 import FormPopup from "../components/PopupWithForm.js";
 import {
   initialCards,
   profileEditButton,
   profileEditModal,
   previewImageModal,
   addCardModal,
+  editAvatarModal,
+  popupConfirmModal,
   profileModalCloseButton,
   addCardModalCloseButton,
   previewImageCloseButton,
   addNewCardButton,
+  avatarEditSelector,
   profileTitle,
   profileDescription,
   profileTitleInput,
   profileDescriptionInput,
   previewImageFooter,
   profileEditForm,
   addCardForm,
+  editAvatarForm,
   cardTemplate,
   cardListEl,
   cardTitleInput,
   cardURLInput,
   validationSettings,
 } from "../utils/constants.js";
+import Api from "../components/Api.js";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "58a7c735-ac44-49af-aeeb-e746dacd4b77",
+    "Content-Type": "application/json",
+  },
+});
+
+const popupConfirm = new PopupConfirm(popupConfirmModal);
+popupConfirm.setEventListeners();
+
+const user = new UserInfo(profileTitle, profileDescription, avatarEditSelector);
+console.log(user);
+
+const renderCard = (cardData) => {
+  const card = new Card(
+    {
+      cardData,
+      userId: userId,
+      handleCardClick: (imageData) => {
+        previewImage.open(imageData);
+      },
+      handleDeleteClick,
+      handleLikeClick,
+    },
+    cardTemplate
+  );
+  cardSection.addItem(card.getCardView());
+};
+
+function handleDeleteClick(item) {
+  popupConfirm.open();
+  popupConfirm.setSubmitAction(() => {
+    api
```

Comment:
Needs correction: don't forget to notify the user about the loading state by changing the submit button text (you should do that for all the forms and the confirmation popup)

Reply:
Needs correction: you should show the loading text before making the api request and hide in `finally` block (fix that for all the api requests)